lsp-metals-protocol.el

```diff
@@ -0,0 +1,36 @@
+;;; lsp-metals-protocol.el --- LSP Metals custom protocol definitions -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2020 Evgeny Kurnevsky <kurnevsky@gmail.com>
+
+;; Version: 1.0.0
+;; Package-Requires: ((emacs "25.2") (lsp-mode "6.0"))
+;; Author: Evgeny Kurnevsky <kurnevsky@gmail.com>
+;; Keywords: languages, extensions
+;; URL: https://github.com/emacs-lsp/lsp-metals
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; lsp-metals custom protocol definitions
+
+;;; Code:
+
+(require 'lsp-protocol)
+
+(lsp-interface
+  (metals/executeClientCommand (:command) (:arguments)))
```

Comment:
this maybe Command from base protocol?

Reply:
It's a custom command with `ExecuteCommandParams` from base protocol: https://scalameta.org/metals/docs/editors/new-editor.html#metalsexecuteclientcommand Can I use it somehow?